### PR TITLE
release workflow: Automatically adjust default release channel

### PR DESCRIPTION
**Pull request by dgn**
This will make sure that the channel field is pre-populated with a good default when we run the release workflow.

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -5,7 +5,8 @@ LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
 LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=sailoperator
-LABEL operators.operatorframework.io.bundle.channels.v1="0.2"
+LABEL operators.operatorframework.io.bundle.channels.v1="0.2-dev"
+LABEL operators.operatorframework.io.bundle.channel.default.v1=0.2-dev
```
Comment (FilipB) [inline bundle.Dockerfile:9]: Is this needed when we have just one channel? I believe this is only useful when having multiple channels in `operators.operatorframework.io.bundle.channel.default.v1`

File: Makefile.core.mk [hunk 2]
```diff
@@ -79,7 +79,12 @@ GINKGO_FLAGS := $(if $(VERBOSE),-v) $(if $(CI),--no-color)
 # To re-generate a bundle for other specific channels without changing the standard setup, you can:
 # - use the CHANNELS as arg of the bundle target (e.g make bundle CHANNELS=candidate,fast,stable)
 # - use environment variables to overwrite this value (e.g export CHANNELS="candidate,fast,stable")
-CHANNELS ?= ${MINOR_VERSION}
+CHANNEL_SUFFIX := dev
```
Comment (FilipB) [inline Makefile.core.mk:82]: Looking at channel naming best [practice](https://olm.operatorframework.io/docs/best-practices/channel-naming/#example-2), this should be probably a prefix.

File: Makefile.core.mk [hunk 3]
```diff
@@ -425,6 +432,9 @@ operator-chart:
 	sed -i -e "s|^\(image: \).*$$|\1${IMAGE}|g" \
 	       -e "s/^\(  version: \).*$$/\1${VERSION}/g" chart/values.yaml
 
+github-workflow:
+	sed -i -e '1,/default:/ s/^\(.*default:\).*$$/\1 ${DEFAULT_CHANNEL}/' .github/workflows/release.yaml
```
Comment (FilipB) [inline Makefile.core.mk:436]: IIUC the `default` field in the workflow actually means a default value for channels which can be list of channels and not a DEFAULT_CHANNEL

File: bundle.Dockerfile (hunk 1 above)
Comment (dgn) [reply bundle.Dockerfile]: good point. removed

File: Makefile.core.mk (hunk 2 above)
Comment (dgn) [reply Makefile.core.mk]: fixed!

File: Makefile.core.mk (hunk 3 above)
Comment (dgn) [reply Makefile.core.mk]: fixed! thank you

**Review by FilipB — APPROVED** [review]

**Pull request merged**